wgsl/index.bs

```diff
@@ -3849,109 +4060,145 @@ Note: Override expressions are a superset of [=const-expressions=].
-    const h = array(a, f, a, f);  // array of AbstractFloat with 4 components: (4.0, 2.0, 4.0, 2.0).
+    // 'row_size' denotes the u32 value 16u.  The type is inferred.
+    let row_size = 16u;
   </xmp>
 </div>
 
 ## `var` Declarations ## {#var-decls}
 
-A <dfn noexport>variable</dfn> is a named reference to memory that can contain a value of a
-particular [=storable=] type.
+A <dfn>variable</dfn> is a named reference to memory that can contain a value
+of a particular [=storable=] type.
 
 Two types are associated with a variable: its [=store type=] (the type of value
 that may be placed in the referenced memory) and its [=reference type=] (the type
 of the variable itself).
-If a variable has store type |T|, [=address space=] |AS|, and [=access mode=] |AM|,
-then its reference type is ref&lt;|AS|,|T|,|AM|&gt;.
-The [=store type=] of a variable is always [=type/concrete=].
-
-A <dfn noexport>variable declaration</dfn>:
-
-* Specifies the variable’s name.
-* Specifies the [=address space=], [=store type=], and [=access mode=].
-    Together these comprise the variable's [=reference type=].
-* Ensures the execution environment allocates memory for a value of the store type, in the specified address space,
-    supporting the given access mode, for the [=lifetime=] of the variable.
-* Optionally has an *initializer* expression, if the variable is in the [=address spaces/private=] or [=address spaces/function=] address spaces.
-    If present, the initializer expression [=shader-creation error|must=] evaluate to the variable's store type.
-
-When an [=identifier=] use [=resolves=] to a variable declaration,
-the identifier is an expression denoting the reference [=memory view=] for the variable's memory,
-and its type is the variable's [=reference type=].
+If a variable has store type `T`, [=address space=] `AS`, and [=access mode=]
+`AM`, then its reference type is `ref<AS,T,AM>`.
+The store type of a variable is always [=type/concrete=].
+
+A variable declaration:
+* Specifies the variable’s [=name=].
+* Determines the variable’s [=address space=], [=store type=], and [=access mode=].
+    Together these comprise the variable’s [=reference type=].
+    * The store type is the [=effective-value-type=] of the variable’s declaration.
+* Ensures the execution environment allocates memory for a value of the store
+    type, in the specified address space, supporting the given access mode, for
+    the [=lifetime=] of the variable.
+* Optionally has an initializer expression if the variable is in the [=address
+    spaces/private=] or [=address spaces/function=] address spaces.
+    If present, the initializer must evaluate to the variable’s store type.
+    If present, the initializer for a [=address spaces/private=] variable must
+    be an [=override-expression=].
+    Variables in other address spaces [=shader-creation error|must not=] have an initializer.
```

Comment:
This reads confusingly., because the "other" might refer to anything other than "private"
Suggest:

> Variables in address spaces other than function or private must not have an initializer.